app/Helpers/Bank/Nordigen/Transformer/TransactionTransformer.php

```diff
@@ -64,85 +64,85 @@
-            (array_key_exists('creditorAccount', $transaction) && array_key_exists('iban', $transaction["creditorAccount"]) ?
-                $transaction['creditorAccount']['iban'] : null);
-        $participant_name = array_key_exists('debtorName', $transaction) ?
-            $transaction['debtorName'] :
-            (array_key_exists('creditorName', $transaction) ?
-                $transaction['creditorName'] : null);
-
-        return [
-            'transaction_id' => $transaction["transactionId"],
-            'amount' => (int) $transaction["transactionAmount"]["amount"],
-            'currency_id' => $this->convertCurrency($transaction["transactionAmount"]["currency"]),
-            'category_id' => null,
-            'category_type' => array_key_exists('additionalInformation', $transaction) ? $transaction["additionalInformation"] : '',
-            'date' => $transaction["bookingDate"],
-            'description' => $description,
-            'participant' => $participant,
-            'participant_name' => $participant_name,
-            'base_type' => (int) $transaction["transactionAmount"]["amount"] <= 0 ? 'DEBIT' : 'CREDIT',
-        ];
+    return $data;
+  }
+
+  public function transformTransaction($transaction)
+  {
+
+    if (!array_key_exists('transactionId', $transaction) || !array_key_exists('transactionAmount', $transaction))
+      throw new \Exception('invalid dataset');
+
+    // description could be in varios places
+    $description = '';
+    if (array_key_exists('remittanceInformationStructured', $transaction))
+      $description = $transaction["remittanceInformationStructured"];
+    else if (array_key_exists('remittanceInformationStructuredArray', $transaction))
+      $description = implode('\n', $transaction["remittanceInformationStructuredArray"]);
+    else if (array_key_exists('remittanceInformationUnstructured', $transaction))
+      $description = $transaction["remittanceInformationUnstructured"];
+    else if (array_key_exists('remittanceInformationUnstructuredArray', $transaction))
+      $description = implode('\n', $transaction["remittanceInformationUnstructuredArray"]);
+    else
+      Log::warning("Missing description for the following transaction: " . json_encode($transaction));
+
+    // participant
+    $participant = array_key_exists('debtorAccount', $transaction) && array_key_exists('iban', $transaction["debtorAccount"]) ?
+      $transaction['debtorAccount']['iban'] :
+      (array_key_exists('creditorAccount', $transaction) && array_key_exists('iban', $transaction["creditorAccount"]) ?
+        $transaction['creditorAccount']['iban'] : null);
+    $participant_name = array_key_exists('debtorName', $transaction) ?
+      $transaction['debtorName'] :
+      (array_key_exists('creditorName', $transaction) ?
+        $transaction['creditorName'] : null);
+
+    return [
+      'transaction_id' => $transaction["transactionId"],
+      'amount' => (int) $transaction["transactionAmount"]["amount"],
```

Comment:
Should this be int or float?

Reply:
@turbo124 you aare right, please check again :)

Reply:
@paulwer 

I don't see this updated yet?

Reply:
@turbo124 its now updated :)